helix-core/src/main/java/org/apache/helix/spectator/RoutingTableSnapshot.java

```diff
@@ -33,9 +36,13 @@
  */
 public class RoutingTableSnapshot {
   private final RoutingTable _routingTable;
+  private final PropertyType _propertyType;
+  private final String _customizedStateType;
 
   public RoutingTableSnapshot(RoutingTable routingTable) {
     _routingTable = routingTable;
+    _propertyType = routingTable.getPropertyType();
```

Comment:
Since you already kept the routingTable, is it still necessary to keep them as private fields?

Reply:
It is used mainly in one of the methods called getCustomizeViews. Based on this type we decide what to return (either empty list or do type casting between routingTable and customizedViewRoutingTable). Please have a look and let me know if that answers your question.